src/xdocs/config_whitespace.xml

```diff
@@ -16,6 +16,266 @@
+          it more readable and adhere to line length standards.
+        </p>
+        <p>
+          Rationale: Long chained method calls are difficult to read and comprehend, and sometimes
+          require horizontal scrolling to read the complete method call chain. Wrapping method calls
+          onto the following line(s) enhances readability.
+        </p>
+      </subsection>
+
+      <subsection name="Properties" id="ChainedMethodCallWrap_Properties">
+        <div class="wrapper">
+          <table>
+            <tr>
+              <th>name</th>
+              <th>description</th>
+              <th>type</th>
+              <th>default value</th>
+              <th>since</th>
+            </tr>
+            <tr>
+              <td>identifierPattern</td>
+              <td>
+                Specify method calls to be checked. By default, identifierPattern matches
+                nothing, so no method calls are checked. This property must be set in order
+                to use the check.
+              </td>
+              <td><a href="property_types.html#Pattern">Pattern</a></td>
+              <td><code>"^$"</code></td>
+              <td>10.4</td>
+            </tr>
+            <tr>
+              <td>maxCallsPerLine</td>
+              <td>
+                  Specify maximum number of method calls per lines if we are wrapping the
+                  method calls. If all the method calls span over a single line then this
+                  property has no effect.
+              </td>
+              <td><a href="property_types.html#int">int</a></td>
+              <td><code>1</code></td>
+              <td>10.4</td>
+            </tr>
+            <tr>
+              <td>maxSingleLineCalls</td>
+              <td>
+                  Specify maximum number of method calls if all chained method calls span
+                  over a single line. If chained method calls span over multiple lines then
+                  this property has no effect.
+              </td>
+              <td><a href="property_types.html#int">int</a></td>
+              <td><code>1</code></td>
+              <td>10.4</td>
+            </tr>
+          </table>
+        </div>
+      </subsection>
+
+      <subsection name="Examples" id="ChainedMethodCallWrap_Examples">
+        <p>
+          To configure the check to use custom identifierPattern to validate assertions:
+        </p>
+        <source>
+&lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ &lt;property name=&quot;identifierPattern&quot; value=&quot;^assert.*$&quot; /&gt;
+&lt;/module&gt;
+        </source>
+        <p>
+          Example:
+        </p>
+        <source>
+void method() {
+    assertWithMessage("ex").that(1).isEqualTo(1);
+                     // violation above, ''3' method calls on single line, max allowed is '1''
+    assertWithMessage("ex")
+        .that(1).isEqualTo(1); // violation, 'Chained method call should be wrapped'
```

Comment:
I am not understanding this violation in general.

There are 2 chained method calls on this line, so why isn't it a size violation when max allowed is 1. Even if we ignore the size, this line is wrapped from the previous line but violation is saying it should be wrapped.

So what does violation mean? Please explain.

Reply:
```java
assertWithMessage("ex").that(1).isEqualTo(1);
                        // violation above, ''3' method calls on single line, max allowed is '1''
```
3 method calls in single line, max allowed is one.

---

```java
assertWithMessage("ex")
    .that(1).isEqualTo(1); // violation, 'Wrap chained method call, max allowed on a line is '1''
```

On the second line, there are two method calls, whereas the rule says, when method chains are wrapped onto the next line, then at max allow 2 methods on a single line.

---

The properties for single-line chain and multi-line chain are determined by properties `maxSingleLineCalls`  and `maxCallsPerLine` respectively. Only one of the property applies for any method chain.

Reply:
It looks like message was changed in new update.
`violation, 'Chained method call should be wrapped'` => `violation, 'Wrap chained method call, max allowed on a line is 1'`.

So issue was just original text and not it makes more sense, so I am resolving.